behind-the-veil-siteroot/imports/ui/components/searchBar/searchBar.jsx

```diff
@@ -4,57 +4,244 @@
+import React, {useEffect, useState} from "react";
+import {MagnifyingGlassIcon, XMarkIcon} from "@heroicons/react/24/outline";
 import classNames from "classnames";
 import "./searchBar.css";
 import Input from "../input/Input";
+import Button from "../button/Button";
+import UrlBasePath from "../../enums/UrlBasePath";
+import {useNavigate} from "react-router-dom";
+import {useServices, useUsers} from "../DatabaseHelper";
+import {useSearchSuggestions} from "../util";
 
 /**
  * General Search Bar component for all required instances of the searching in the app.
  * Works with enter button on keyboard (Requires a separate button for users who use mouse click)
  * @param className custom classes for the search input field
- * @param placeholderName the placeholder name for the search input field (i.e. what the user will see if the input field is empty)
- * @param value the value of the input field (whatever the customer inputs)
- * @param onChange the function that occurs when the search input field changes
- * @param handleSubmit the function which dictates what happens when a user submits the value in the search input field
+ * @param defaultType
+ * @param placeholder
+ * @param startingValue
+ * @param suggestionsDown
  * @param searchBarProps encompasses all other props supplied and applies them to the search input field
  */
 const SearchBar = ({
-                       className,
-                       placeholderName,
-                       value,
-                       onChange,
-                       handleSubmit,
+                       classnames,
+                       defaultType = 'services',
+                       placeholder,
+                       startingValue,
+                       suggestionsDown,
                        ...searchBarProps
                    }) => {
+    let navigate = useNavigate();
+
+    // if type is not a valid value, default to services
+    if (defaultType !== 'services' && defaultType !== 'artists') {
+        defaultType = 'services'
+    }
+
+    // creates a state of the input value for the search bar
+    const [inputValue, setInputValue] = useState(startingValue);
+    const [searchType, setSearchType] = useState(defaultType);
+
+    // load data
+    const userFilter = {"profile.type": "artist"}
+    const {isLoading: isLoadingArtists, usersData} = useUsers("all_artists", [], userFilter)
+
+    const serviceFilter = {serviceActive: true}
+    const {isLoading: isLoadingServices, servicesData} = useServices("active_services", [], serviceFilter, true)
+
+    // variables set up for suggestions
+    const [fullSuggestions, setFullSuggestions] = useState([]);
+    const [loadedFullSuggestions, setLoadedFullSuggestions] = useState(false);
+    const [filteredSuggestions, setFilteredSuggestions] = useState([]);
+    const [selectedSuggestions, setSelectedSuggestions] = useState(false);
+    const [maxSuggestion, setMaxSuggestion] = useState(10);
```

Comment:
setMaxSuggestion is never called, maxSuggestion shouldn't be a state, it can just be a constant
`MAX_SUGGESTION  = 10`